experimental/sdk/webpubsub-socketio-extension/src/SIO/index.ts

```diff
@@ -32,9 +36,32 @@ export async function useAzureSocketIO(
   // `attachServe` is a Socket.IO design which attachs static file serving to internal http server.
   // Creating new engine makes previous `attachServe` execution invalid.
   // Reference: https://github.com/socketio/socket.io/blob/4.6.2/lib/index.ts#L518
+  debug("serve static file");
   if (this["_serveClient"]) {
-    this["attachServe"](this["httpServer"]);
+    this["attachServe"](httpServer);
   }
+
+  // Add negotiate handler
+  debug("add negotiate handler");
+
+  const path = this._opts.path || "/socket.io";
+  const negotiatePathPrefix = path + (path.endsWith("/") ? "" : "/") + "negotiate";
+
+  const listeners = httpServer.listeners("request").slice(0);
+  httpServer.removeAllListeners("request");
+  httpServer.on("request", (req: IncomingMessage, res: ServerResponse) => {
+    if (negotiatePathPrefix === req.url.slice(0, negotiatePathPrefix.length)) {
+      if (!webPubSubOptions.negotiate) {
+        webPubSubOptions.negotiate = getDefaultNegotiateHandler(engine);
+      }
+      webPubSubOptions.negotiate(req, res, engine.webPubSubConnectionManager.service.getClientAccessTokenUrl);
+    } else {
+      for (let i = 0; i < listeners.length; i++) {
+        listeners[i].call(httpServer, req, res);
```

Comment:
`/socket.io` requests still call these listeners?

Reply:
resolved this issue